src/pymedphys/gamma/_level2/gammashell.py

```diff
@@ -116,160 +114,290 @@ def gamma_shell(coords_reference, dose_reference,
         coords_reference, dose_reference,
-        coords_evaluation, dose_evaluation)
-
-    if global_normalisation is None:
-        global_normalisation = np.max(dose_reference)
-
-    global_dose_threshold = dose_percent_threshold / 100 * global_normalisation
-    lower_dose_cutoff = lower_percent_dose_cutoff / 100 * global_normalisation
-
-    if not quiet:
-        if local_gamma:
+        coords_evaluation, dose_evaluation,
+        dose_percent_threshold, distance_mm_threshold,
+        lower_percent_dose_cutoff, interp_fraction,
+        max_gamma, local_gamma,
+        global_normalisation, skip_once_passed,
+        random_subset,
+        ram_available, quiet)
+
+    if not options.quiet:
+        if options.local_gamma:
             print('Calcing using local normalisation point for gamma')
         else:
             print('Calcing using global normalisation point for gamma')
-        print('Global normalisation set to {} Gy'.format(global_normalisation))
+        print('Global normalisation set to {} Gy'.format(
+            options.global_normalisation))
         print('Global dose threshold set to {} Gy ({}%)'.format(
-            global_dose_threshold, dose_percent_threshold))
-        print('Distance threshold set to {} mm'.format(distance_mm_threshold))
+            options.global_dose_threshold, options.dose_percent_threshold))
+        print('Distance threshold set to {} mm'.format(
+            options.distance_mm_threshold))
         print('Lower dose cutoff set to {} Gy ({}%)'.format(
-            lower_dose_cutoff, lower_percent_dose_cutoff))
+            options.lower_dose_cutoff, lower_percent_dose_cutoff))
         print('')
 
-    distance_step_size = distance_mm_threshold / interp_fraction
-    maximum_test_distance = distance_mm_threshold * max_gamma
+    current_gamma = gamma_loop(options)
 
-    evaluation_interpolation = RegularGridInterpolator(
-        coords_evaluation, np.array(dose_evaluation),
-        bounds_error=False, fill_value=np.inf
-    )
+    gamma = {}
+    for i, dose_threshold in enumerate(options.dose_percent_threshold):
+        for j, distance_threshold in enumerate(options.distance_mm_threshold):
+            key = (dose_threshold, distance_threshold)
+
+            gamma_temp = current_gamma[:, i, j]
+            gamma_temp = np.reshape(
+                gamma_temp, np.shape(dose_reference))
+            gamma_temp[np.isinf(gamma_temp)] = np.nan
+
+            with np.errstate(invalid='ignore'):
+                gamma_greater_than_ref = gamma_temp > max_gamma
+                gamma_temp[gamma_greater_than_ref] = max_gamma
+
+            gamma[key] = gamma_temp
+
+    if not options.quiet:
+        print('\nComplete!')
+
+    if len(gamma.keys()) == 1:
+        gamma = next(iter(gamma.values()))
+
+    return gamma
+
+
+def default_ram() -> int:
+    memory = psutil.virtual_memory()
+    total_memory = memory.total
+
+    ram_available = int(total_memory * 0.8)
```

Comment:
I was thinking this is rather high. And why not look at availability directly instead of a fraction of total memory?

Reply:
Would changing it to something like `0.3 * total_memory work` better? The confusing thing is many OSs will use RAM as a cache, so sometimes quite a bit of RAM is available but the OS just needs to have that RAM asked for...

Reply:
For now I just changed the default to `0.3 * total_memory`.

Reply:
For me personally, 0.3 as a default would improve performance. And it's probably the case that a majority of users are in a similar situation - lots of institutional IT-enforced RAM use...